aws/resource_aws_wafv2_web_acl_test.go

```diff
@@ -0,0 +1,1120 @@
+					"rule_group_reference_statement": []interface{}{
+						map[string]interface{}{
+							"arn":           aws.StringValue(r.Rules[0].Statement.RuleGroupReferenceStatement.ARN),
+							"excluded_rule": e,
+						},
+					},
+					"size_constraint_statement": []interface{}{},
+					"sqli_match_statement":      []interface{}{},
+					"xss_match_statement":       []interface{}{},
+				},
+			},
+			"visibility_config": []interface{}{
+				map[string]interface{}{
+					"cloudwatch_metrics_enabled": false,
+					"metric_name":                "friendly-rule-metric-name",
+					"sampled_requests_enabled":   false,
+				},
+			},
+		}
+
+		f := schema.HashResource(ruleResource)
+		*idx = f(rule)
+
+		return nil
+	}
+}
+
+func testAccCheckAwsWafv2WebACLDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_wafv2_web_acl" {
+			continue
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).wafv2conn
+		resp, err := conn.GetWebACL(
+			&wafv2.GetWebACLInput{
+				Id:    aws.String(rs.Primary.ID),
+				Name:  aws.String(rs.Primary.Attributes["name"]),
+				Scope: aws.String(rs.Primary.Attributes["scope"]),
+			})
+
+		if err == nil {
+			if resp == nil || resp.WebACL == nil {
+				return fmt.Errorf("Error getting WAFv2 WebACL")
+			}
+			if aws.StringValue(resp.WebACL.Id) == rs.Primary.ID {
+				return fmt.Errorf("WAFv2 WebACL %s still exists", rs.Primary.ID)
+			}
+		}
+
+		// Return nil if the WebACL is already destroyed
+		if isAWSErr(err, wafv2.ErrCodeWAFNonexistentItemException, "") {
+			return nil
+		}
+
+		return err
+	}
+
+	return nil
+}
+
+func testAccCheckAwsWafv2WebACLExists(n string, v *wafv2.WebACL) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No WAFv2 WebACL ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).wafv2conn
+		resp, err := conn.GetWebACL(&wafv2.GetWebACLInput{
+			Id:    aws.String(rs.Primary.ID),
+			Name:  aws.String(rs.Primary.Attributes["name"]),
+			Scope: aws.String(rs.Primary.Attributes["scope"]),
+		})
+
+		if err != nil {
+			return err
+		}
+
+		if resp == nil || resp.WebACL == nil {
+			return fmt.Errorf("Error getting WAFv2 WebACL")
+		}
+
+		if aws.StringValue(resp.WebACL.Id) == rs.Primary.ID {
+			*v = *resp.WebACL
+			return nil
+		}
+
+		return fmt.Errorf("WAFv2 WebACL (%s) not found", rs.Primary.ID)
+	}
+}
+
+func testAccAwsWafv2WebACLConfig_Basic(name string) string {
+	return fmt.Sprintf(`
+resource "aws_wafv2_web_acl" "test" {
+  name        = "%s"
+  description = "%s"
+  scope       = "REGIONAL"
+
+  default_action {
+    allow {}
+  }
+
+  visibility_config {
+    cloudwatch_metrics_enabled = false
+    metric_name                = "friendly-metric-name"
+    sampled_requests_enabled   = false
+  }
+}
+`, name, name)
```

Comment:
Since the parameters are repeated, could we use argument indexes, i.e. `%[1]s` in `fmt.Sprintf()` instead?